### PR TITLE
fix(job-server) store error class and stack trace in database

### PR DESCRIPTION
**Pull request by krcz**
* Implement storing and fetching error class and stack trace in DAOs
* Create migrations adding columns in SQL databases
* Modify tests to take care of new columns

BREAKING CHANGES :
* Schema change: new columns in SQL databases and Cassandra
* File storage format incompatible
* Stack trace returned as string instead of list of strings

**Pull Request checklist**

- [*] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?
- [*] Tests for the changes have been added (for bug fixes / features) ?
- [*] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :**
Spark JobServer doesn't store error stacktrace and class, returning java.lang.Throwable and stack trace pointing to JobDao read when getting historic job: #901 

**New behavior :**
Spark JobServer stores error class and stacktrace and returns these for historic jobs.

**BREAKING CHANGES**
If Cassandra storage is used, new columns must be created manually.
Format of stack trace is no longer a list, but a string, standard Java exception printed stack trace. Cause is not a separate field in REST response JSON, but is embedded in stack trace.

**Other information**:

---
This change is [Reviewable](https://reviewable.io/reviews/spark-jobserver/spark-jobserver/923)

### CODE REVIEW

**Comment by velvia** [comment]
Thanks for this, great idea

File: job-server/src/main/resources/db/h2/migration/V0_7_4/V0_7_4__create_error_class_and_stacktrace.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE JOBS ADD COLUMN ERROR_CLASS VARCHAR(255);
```
[the remark this answers is not in the document]
Comment (krcz) [reply job-server/src/main/resources/db/h2/migration/V0_7_4/V0_7_4__create_error_class_and_stacktrace.sql]: That might cause a problem, but CLASSPATH column has the same limit. I can change it here to TEXT / CLOB if you want, but that might be an overkill.

**Comment by krcz** [comment]
@velvia stack value is now text and contains the same content that is printed to the output on uncaught exception / in logging and cotains the cause (if available), and even the cause of the cause (if available as well):
https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#printStackTrace()

 The only loss is we won't get a full stack for the cause, only the last few calls.

File: job-server/src/main/resources/db/h2/migration/V0_7_4/V0_7_4__create_error_class_and_stacktrace.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (velvia) [reply job-server/src/main/resources/db/h2/migration/V0_7_4/V0_7_4__create_error_class_and_stacktrace.sql]: Nah that's ok

**Comment by velvia** [comment]
@krcz thanks, let's merge and then we can adjust further.  Need to run it and see in practice how much of the cause is kept.

**Pull request merged**